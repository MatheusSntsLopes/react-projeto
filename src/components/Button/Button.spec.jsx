import { Button} from '.';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
describe('<Button text="load more" />', ()=> {
    it('should render the button with the text "Load more"', ()=> {
        render(<Button text="Load more" />);

        const button = screen.getByRole('button', { name: /load more/i });
        expect(button).toBeInTheDocument();
    });

    it('should call function on button click', ()=> {
        const fn = jest.fn();
        render(<Button text="Load more" onClick={fn} />);

        const button = screen.getByRole('button', { name: /load more/i });

       
        userEvent.click(button);
        expect(fn).toHaveBeenCalledTimes(1);
    });

    it('should be disable when disabled is true', ()=> {
        
        render(<Button text="Load more" disabled={true} />);

        const button = screen.getByRole('button', { name: /load more/i });

       expect(button).toBeDisabled();
    });

    it('should match snapshot', ()=> {
        
       const {container} = render(<Button text="Load more" disabled={true} />);

       expect(container).toMatchSnapshot();
    });
})