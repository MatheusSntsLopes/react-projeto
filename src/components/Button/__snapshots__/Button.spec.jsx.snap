// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button text="load more" /> should match snapshot 1`] = `
<div>
  <button
    class="button"
    disabled=""
  >
    Load more
  </button>
</div>
`;
